src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -122,9 +122,17 @@ private static void ReportPartialError(Location errorLocation, DiagnosticBag dia
 
         internal static DeclarationModifiers AdjustModifiersForAnInterfaceMember(DeclarationModifiers mods, bool hasBody, bool isExplicitInterfaceImplementation)
         {
-            if ((mods & (DeclarationModifiers.Static | DeclarationModifiers.Private | DeclarationModifiers.Partial | DeclarationModifiers.Virtual | DeclarationModifiers.Abstract)) == 0 &&
-                !isExplicitInterfaceImplementation)
+            if (isExplicitInterfaceImplementation)
             {
+                if ((mods & DeclarationModifiers.Abstract) != 0)
+                {
+                    mods |= DeclarationModifiers.Sealed;
+                }
+            }
+            else if ((mods & (DeclarationModifiers.Static | DeclarationModifiers.Private | DeclarationModifiers.Partial | DeclarationModifiers.Virtual | DeclarationModifiers.Abstract)) == 0)
+            {
+                Debug.Assert(!isExplicitInterfaceImplementation);
+
```

Comment:
Given that this is in the `else` of an `if (isExplicitInterfaceImplementation)`, the assert seems unnecessary.

Reply:
> Given that this is in the else of an if (isExplicitInterfaceImplementation), the assert seems unnecessary.

It is here for the benefit of the reader and to prevent unintentional break in the future.

---
In reply to: [285293830](https://github.com/dotnet/roslyn/pull/35756#discussion_r285293830) [](ancestors = 285293830)